docs/source/export.rst

```diff
@@ -0,0 +1,65 @@
+file and not the read file.
+
+- If the modifications are removals or additions of containers, then no special action must be taken, as long as the
+  container hierarchy is updated correspondingly.
+- If the modifications are changes to attributes, then
+  :py:meth:`Container.set_modified() <hdmf.container.AbstractContainer.set_modified>` must be called
+  on the container before exporting.
+
+.. note::
+
+  Modifications to :py:class:`h5py.Dataset <h5py.Dataset>` objects act *directly* on the read file on disk.
+  Changes are applied immediately and do not require exporting or writing the file.
+
+Can I export a newly instantiated container?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+No, you can only export containers that have been read from a file. The ``src_io`` argument is required in
+:py:meth:`HDMFIO.export <hdmf.backends.io.HDMFIO.export>`.
+
+Can I read a container from disk and export only part of the container?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+It depends. You can only export the root container from a file. To export the root container without certain other
+sub-containers in the hierarchy, you can remove those other containers before exporting. However, you cannot export
+only a sub-container of the container hierarchy.
+
+What happens to links when I export?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The exported file will not contain any links to the original file.
+
+All links (such as internal links (i.e., HDF5 soft links) and links to other files (i.e., HDF5 external links))
+will be preserved in the exported file.
+
+If a link to an :py:class:`h5py.Dataset <h5py.Dataset>` in another file is added to the in-memory container after
+reading it from file and then exported, then by default, the export process will create an external link to the
+existing :py:class:`h5py.Dataset <h5py.Dataset>` object. To instead copy the data from the
+:py:class:`h5py.Dataset <h5py.Dataset>` in another
+file to the exported file, pass the keyword argument ``write_args={'link_data': False}`` to
+:py:meth:`HDF5IO.export <hdmf.backends.hdf5.h5tools.HDF5IO.export>`. This is similar to passing the keyword argument
+``link_data=False`` to :py:meth:`HDF5IO.write <hdmf.backends.hdf5.h5tools.HDF5IO.write>` when writing a file with a
+copy of externally linked datasets.
+
+What happens to references when I export?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+References will be preserved in the exported file.
+NOTE: Exporting a file involves loading into memory all datasets that contain references and attributes that are
+references. The HDF5 reference IDs within an exported file may differ from the reference IDs in the original file.
+
+Can I write a newly instantiated container to two different files?
```

Comment:
This section should be moved up so that the ``Can I do X`` sections appear together and the ``What happens to Y`` section appear together. It may also be useful to divide these into two sections with these as subsections.